tests/experimental/pydantic/test_basic.py

```diff
@@ -272,6 +272,46 @@ class UserType:
     assert field3.type.of_type is str
 
 
+def test_default_and_default_factory():
+    class User1(pydantic.BaseModel):
+        friend: Optional[str] = "friend"
+
+    @strawberry.experimental.pydantic.type(User1)
+    class UserType1:
+        friend: strawberry.auto
+
+    assert UserType1().friend == "friend"
+
+    class User2(pydantic.BaseModel):
+        friend: Optional[str] = None
+
+    @strawberry.experimental.pydantic.type(User2)
+    class UserType2:
+        friend: strawberry.auto
+
+    assert UserType2().friend is None
+
+    # Test instantiation using default_factory
+
+    class User3(pydantic.BaseModel):
+        friend: Optional[str] = pydantic.Field(default_factory=lambda: "friend")
+
```

Comment:
previously defining it like this would fail, because default=None, and we'll think that a factory and default was both defined. 
see https://discord.com/channels/689806334337482765/773519351423827978/927667160250212462